test/media/streaming_engine_unit.js

```diff
@@ -3041,6 +3041,33 @@ describe('StreamingEngine', () => {
       expect(secondRequest.abort).toHaveBeenCalled();
     });
 
+    it('aborts segment without size', async () => {
```

Comment:
Wait, hold on, this test is exactly the same as `still aborts if previous segment size unknown`. The only difference is the comments. The name even sounds like it's supposed to be testing the same thing.
Which probably means that that test isn't working as intended also.

Reply:
Yes it's the same, i tried to investigate the problem, but i didn't found it :(

Reply:
@theodab, do you have been able to see why the test is not working?

Reply:
Sorry for the delay. I got wrapped up in other work.

I believe I have found the problem here.
That existing test, `still aborts if previous segment size unknown`, is testing to see if you can abort when adapting OUT of a variant with unknown segment size.
But that doesn't test to see if you can abort when adapting INTO a variant with unknown segment size. And adding that functionality is what this code is doing.
So having a separate test could be worthwhile after all.

You'll want to modify the test a bit, though. I'll highlight the areas that will need changes.
For starters, this test should probably have a name that highlights this distinction better... perhaps `still aborts if new segment size unknown`, to mirror the other test?